frontend/store/collaterals.ts

```diff
@@ -98,12 +100,17 @@ export const actions = {
                     priceDropRatio: error.toString(),
                 };
             });
+            const autoRouteData = await getUniswapAutoRoute(network, collateral.symbol);
+            const autoRouteExchanges = autoRouteData.route[0].tokenPath.map(p => p.symbol);
```

Comment:
Please make this function as well as the extraction of the route safe. In case there is no connection/no token/no tokenPath, etc this whole page should still work. Suggestion: extract required parameters _inside_ `getUniswapAutoRoute`, try/catch it here (see `.catch` all over the place here)